branchpro/models.py

```diff
@@ -42,3 +43,129 @@ def simulate(self, parameters, times):
             value appears in ``times``.
         """
         raise NotImplementedError
+
+#
+# BranchProModel Class
+#
+
+
+class BranchProModel(ForwardModel):
+    """BranchProModel Class:
+    Class for the models following a Branching Processes behaviour.
+    It inherits from the `ForwardModel`` class.
+
+    Parameters
+    ----------
+    initial_r
+        (numeric) Value of the recombination number at the beginning
+        of the epidemic.
+    serial_interval:
+        (list) Unnormalised probability distribution of that the recipient
+        first displays symptoms s days after the infector first displays
+        symptoms.
+
+    Methods
+    -------
+    simulate: return model output for specified parameters and times.
+    _normalised_daily_mean: (Private) returns the expected number of new cases
+        at time t.
+    """
+
+    def __init__(self, initial_r, serial_interval):
+        super(BranchProModel, self).__init__()
+
+        if not isinstance(serial_interval, list):
+            raise TypeError('Serial interval values must be in a list format.')
+        if np.sum(serial_interval) <= 0:
+            raise ValueError('Sum of serial interval values must be > 0.')
+        if not isinstance(initial_r, (int, float)):
+            raise TypeError('Value of R must be integer or float.')
```

Comment:
```suggestion
            raise TypeError(f'Value of R must be integer or float but got {type(initial_r)}.')
```

Reply:
I see what you mean, and I do not disagree that it is useful, just isn't it a bit of an overkill? Not saying I cannot do it.